Solution1/Solution1.Tests/BL/Services/WeatherApiServiceTests.cs

```diff
@@ -56,5 +58,83 @@ public async Task GetByCityNameAsync_ReturnedWeatherApiDTO_Success()
+            var cityName = "Minsk";
+            var lat = 53;
+            var lon = 27;
+            var urlCoordinates = $"http://api.openweathermap.org/geo/1.0/direct?q={cityName}&appid=3fe39edadae3ae57d133a80598d5b120";
+            var urlForecast = $"https://api.openweathermap.org/data/2.5/forecast?lat={lat}&lon={lon}&cnt=2&units=metric&appid=3fe39edadae3ae57d133a80598d5b120";
+            
+            var responseCoordinates = new HttpResponseMessage
+            {
+                StatusCode = HttpStatusCode.OK,
+                Content = new StringContent(@"[{""name"": ""Minsk"", ""lat"" : 53, ""lon"" : 27}]"),
+            };
+            var responseForecast = new HttpResponseMessage
+            {
+                StatusCode = HttpStatusCode.OK,
+                Content = new StringContent(@"{""city"":
+                                                    {
+                                                        ""name"":""Minsk""
+                                                    },
+                                                ""list"":
+                                                    [
+                                                        {""dt_txt"":""2022-03-05 18:00:00"",
+                                                        ""main"":{""temp"":2}},
+                                                        {""dt_txt"":""2022-03-05 21:00:00"",
+                                                        ""main"":{""temp"":4}}
+                                                    ]}"),
+            };
+
+            _httpMessageHandler
+               .Protected()
+               .Setup<Task<HttpResponseMessage>>(
+                  "SendAsync",
+                  ItExpr.Is<HttpRequestMessage>(
+                      request =>
+                      request.Method == HttpMethod.Get
+                      && request.RequestUri.ToString() == urlCoordinates),
+                  ItExpr.IsAny<CancellationToken>())
+               .ReturnsAsync(responseCoordinates);
+
+            _httpMessageHandler
+               .Protected()
+               .Setup<Task<HttpResponseMessage>>(
+                  "SendAsync",
+                  ItExpr.Is<HttpRequestMessage>(
+                      request =>
+                      request.Method == HttpMethod.Get
+                      && request.RequestUri.ToString() == urlForecast),
+                  ItExpr.IsAny<CancellationToken>())
+               .ReturnsAsync(responseForecast);
+            
+            // Act
+            var result = await _weatherApiService.GetForecastByCityNameAsync(cityName, 2);
+
+            // Assert
+            var expected = new ForecastWeatherApiDTO()
+            {
+                City = new CityApiDTO() { Name = cityName },
+                WeatherPoints = new List<WeatherInfoApiDTO>()
+                {
+                    new WeatherInfoApiDTO()
+                    {
+                        DateTime = new DateTime(2022, 03, 05, 18, 00, 00),
```

Comment:
if you use same values in tests - in arrange, assert and so on, better to define vars, cause if you want to change smth, you won't do it in several places. It's about ALL tests, not only this place